metrics/honeycomb.go

```diff
@@ -281,10 +276,11 @@ func average(vals []float64) float64 {
 }
 
 func (h *HoneycombMetrics) Register(name string, metricType string) {
+	h.lock.Lock()
```

Comment:
there is a lock acquired in the parent function. I _think_ if parent does a Read lock and here it does a Write lock, it will not deadlock, but needs to be verified

Reply:
this was helpful in my past journey with deadlocks: https://github.com/sasha-s/go-deadlock

Reply:
Yup, you're right. I've refactored to use a RWLock and made sure we release the read lock before attempting to get a write lock.